src/content/en/updates/2019/08/native-file-system.md

```diff
@@ -0,0 +1,457 @@
+in a picker, after which a [`FileSystemDirectoryHandle`][fs-dir-handle]
+is returned, which lets you enumerate and access the directory’s files.
+
+```js
+const butDir = document.getElementById('butDirectory');
+butDir.addEventListener('click', async (e) => {
+  const opts = {type: 'openDirectory'};
+  const handle = await window.chooseFileSystemEntries(opts);
+  const entries = await handle.getEntries();
+  for await (const entry of entries) {
+    const kind = entry.isFile ? 'File' : 'Directory';
+    console.log(kind, entry.name);
+  }
+});
+```
+
+## What’s currently supported? {: #whats-supported }
+
+We’re still working on some of the implementation for the
+Native File System API, and not everything in the spec (or explainer)
+has been completed.
+
+As of Chrome 78, the following functionality is not available, or
+doesn't match the spec:
+
+* Handles are not serializable, meaning they cannot be passed via
+  `postMessage()`, or stored in IndexedDB.
+* Non-atomic writes (i.e. calls to `FileSystemFileHandle.createWriter()`
+  with `inPlace: true`).
+* Writing to a file using a [`WritableStream`][writablestream].
+* The [`FileSystemDirectoryHandle.resolve()`][fs-dir-handle] method.
+
+## Security and permissions {: #security-considerations }
+
+<style>
+  figcaption {font-size: smaller;}
+</style>
+
+We’ve designed and implemented the Native File System using the core
+principles we defined in
+[Controlling Access to Powerful Web Platform Features][powerful-apis],
+including user control and transparency, and user ergonomics.
+
+### Opening a file or saving a new file
+
+<figure class="attempt-right">
+  <a href="/web/updates/images/2019/08/fs-open.jpg">
+    <img src="/web/updates/images/2019/08/fs-open.jpg">
+  </a>
+  <figcaption>
+    File picker used to open an existing file for reading.
+  </figcaption>
+</figure>
+
+When opening a file, the user provides permission to read a file or
+directory via the file picker. The open file picker can only be shown via
+a user gesture when served from a [secure context][secure-contexts]. If
+the user changes their mind, they can cancel the selection in the file
+picker and the site does not get access to anything. This is the same
+behavior as the `<input type="file">` element.
+
+<div class="clearfix"></div>
+
+<figure class="attempt-left">
+  <a href="/web/updates/images/2019/08/fs-save.jpg">
+    <img src="/web/updates/images/2019/08/fs-save.jpg">
+  </a>
+  <figcaption>
+    File picker used to save a file to disk.
+  </figcaption>
+</figure>
+
+Similarly, when a web app wants to save a new file, the browser will show
+the save file picker, allowing the user to specify the name and location
+of the new file. Since they are saving a new file to the device (versus
+overwriting an existing file), the file picker grants the app permission
+to write to the file.
+
+<div class="clearfix"></div>
+
+#### Restricted folders
+
+To help protect users and their data, the browser may limit the user’s
```

Comment:
`s/limit the user’s/limit the` to avoid having "users" and "user's" in the same phrase